portfolio/src/main/java/com/google/sps/servlets/StoreDataServlet.java

```diff
@@ -35,78 +38,96 @@
     response.getWriter().println(imageUrl);
 
-    // Do not store an empty comment
-    if(comment.length() == 0) {
+    // Do not store an empty comment,redirect the client to the contact.html page and exit the function
+    if (comment.length() == 0) {
       response.sendRedirect("/contact.html");
       return;
     }
+    // store the client's email address alongside their comments for better querying of individual comments
     Entity commentEntity = new Entity("Comments");
     commentEntity.setProperty("name",username);
     commentEntity.setProperty("comment",comment);
     commentEntity.setProperty("imageUrl",imageUrl);
+    commentEntity.setProperty("emailAddress", emailAddress);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect("/contact.html");
   }
 
   /**
-  * Generate the user details that make up a comment sent by the user.
-  */
+   * Generate the client's details that make up a comment sent by the client.
+   * @param request This holds the request details from the client
+   * @param name This is the name given to the input field on the client's side
+   * @param defaulValue This holds the default value to return in case the clients value is empty
+   * @return the client's value or the default value
+   */
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
     return value.length() == 0 ? defaultValue : value;
   }
 
-  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  /** 
+   * Genereta a URL that points to the uploaded file. 
+   * @param request This holds the client's request details
+   * @param formInputElementName This holds the name of the form which contains the clients file input field
+   * @return the URL that points to the uploaded file or null if the client didn't upload a file
+   */
   private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
     BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
     Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
-    List<BlobKey> blobKeys = blobs.get("image");
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
 
-    // User submitted form without selecting a file, so we can't get a URL. (dev server)
-    if (blobKeys == null || blobKeys.isEmpty()){
+    // Client submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
       return null;
     }
 
     // Our form only contains a single file input, so get the first index.
     BlobKey blobKey = blobKeys.get(0);
 
-    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    // Client submitted form without selecting a file, so we can't get a URL. (live server)
```

Comment:
I think this comment should go above the if statement below (line 115) instead of here